relayer/crates/starknet-chain-components/src/impls/types/address.rs

```diff
@@ -36,7 +36,7 @@ impl<Chain: Async> ProvideAddressType<Chain> for ProvideFeltAddressType {
     Deserialize,
 )]
 #[display("0x{_0:x}")]
-pub struct StarknetAddress(Felt);
+pub struct StarknetAddress(pub Felt);
```

Comment:
why do you need `pub` here? We already impl `Deref`.

Reply:
The preferred approach for CGP is to use generics instead of newtypes with private fields to hide internal details. Private fields in crates like ibc-rs have been problematic especially in Rust, due to the excessive needs to provide all kinds of utility methods at the definition side to deal with all possible ways of interaction, e.g. owned/ref/mut/from etc.

Especially in this case, there is no invariant or added benefit of hiding the fact that a Starknet address is a `Felt`. The only reason we have it is to implement a custom `Display` instance, and the only reason we use newtype is that `Display` is not a CGP trait.

Reply:
It's opinionated, I see 😅 Since, we are using hermes-sdk and cgp, it's ok.